hook-common/src/kafka_messages/app_metrics.rs

```diff
@@ -0,0 +1,123 @@
+use chrono::{DateTime, Utc};
+use serde::{Serialize, Serializer};
+use uuid::Uuid;
+
+use super::{serialize_datetime, serialize_uuid};
+
+#[derive(Serialize)]
+pub enum AppMetricCategory {
+    ProcessEvent,
+    OnEvent,
+    ScheduledTask,
+    Webhook,
+    ComposeWebhook,
+}
+
+#[derive(Serialize)]
+pub enum ErrorType {
+    Timeout,
+    Connection,
+    HttpStatus(u16),
+}
+
+#[derive(Serialize)]
+pub struct ErrorDetails {
+    pub error: Error,
+    // TODO: The plugin-server sends the entire raw event with errors. In order to do this, we'll
+    // have to pass the entire event when we enqueue items, and store it in the Parameters JSONB
+    // column. We should see if it's possible to work around this before we commit to it.
```

Comment:
As we discussed, we are only sticking with an error message as part of the details, so I wouldn't include event as a field.

Reply:
I forgot about this comment, I do think we should probably add the event ID here, so we have some way to tie it back to the event.

Reply:
It doesn't have to be now or anything, though.

Reply:
Yeah, metadata can be extended with `event_id` :+1: